### PR TITLE
[cubeb] Add new port

### PR DESCRIPTION
**Pull request by Zopolis4**
**Describe the pull request**

- #### What does your PR do?
  Adds [cubeb](https://github.com/mozilla/cubeb) port

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?
  Everything except UWP, No

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?
  Yes

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?
  Yes

### CODE REVIEW

File: ports/cubeb/vcpkg.json [hunk 1]
```diff
@@ -0,0 +1,17 @@
+{
+  "name": "cubeb",
+  "version-string": "dc511c6b3597b6384d28949285b9289e009830ea",
```
Comment (FrankXie05) [inline ports/cubeb/vcpkg.json:3]: For the version field, see https://github.com/microsoft/vcpkg/blob/master/docs/users/versioning.md

File: ports/cubeb/portfile.cmake [hunk 2]
```diff
@@ -0,0 +1,17 @@
+vcpkg_from_github(
+    OUT_SOURCE_PATH SOURCE_PATH
+    REPO mozilla/cubeb
+    REF dc511c6b3597b6384d28949285b9289e009830ea
+    SHA512 a4ccd3f0a156db4e2e75a8d231e95a08d555390571551cb3e92c71cdee46dc74dc66b5272fda4b5f1f083b92672b360152cefd38be242f238fe802acc1ea17e9
+    HEAD_REF master
+)
+
+vcpkg_cmake_configure(SOURCE_PATH ${SOURCE_PATH} OPTIONS -DUSE_SANITIZERS=OFF -DBUILD_TESTS=OFF)
```
Comment (FrankXie05) [inline ports/cubeb/portfile.cmake:9]:
```suggestion
vcpkg_cmake_configure(
    OURCE_PATH "${SOURCE_PATH}"
    OPTIONS 
        -DUSE_SANITIZERS=OFF 
        -DBUILD_TESTS=OFF
)
```

**Review by FrankXie05 — CHANGES_REQUESTED** [review]

File: versions/c-/cubeb.json [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+    {
+      "git-tree": "c5551b587ba2523802e7c732a9a3673eaaa60e5b",
+      "version-date": "2022-07-22",
+      "port-version": 0
+    },
+    {
+      "git-tree": "68534941d46e3b7e2c65e7bf7768fe3de2afa250",
+      "version-string": "dc511c6b3597b6384d28949285b9289e009830ea",
+      "port-version": 0
+    }
```
Comment (FrankXie05) [inline versions/c-/cubeb.json:12]:
```suggestion
```

**Review by FrankXie05 — CHANGES_REQUESTED** [review]

**Comment by FrankXie05** [comment]
Log form `install-arm-uwp-dbg-out.log`:
````
D:\buildtrees\cubeb\src\9e009830ea-4cc11e5bd0.clean\src\cubeb_winmm.c(102): error C2061: syntax error: identifier 'WAVEHDR'
D:\buildtrees\cubeb\src\9e009830ea-4cc11e5bd0.clean\src\cubeb_winmm.c(109): error C2061: syntax error: identifier 'waveout'
````
You can get the latest error log [here](https://dev.azure.com/vcpkg/public/_build/results?buildId=77342&view=artifacts&pathAsName=false&type=publishedArtifacts)
[install-arm-uwp-dbg-out.log](https://github.com/microsoft/vcpkg/files/9476668/install-arm-uwp-dbg-out.log)

**Comment by Zopolis4** [comment]
The UWP build problem is something I have encountered before and have reported at mozilla/cubeb#721. I'll update the port if it gets fixed, but for now I've just marked it as incompatible.

File: ports/cubeb/portfile.cmake [hunk 4]
```diff
@@ -0,0 +1,23 @@
+vcpkg_from_github(
+    OUT_SOURCE_PATH SOURCE_PATH
+    REPO mozilla/cubeb
+    REF dc511c6b3597b6384d28949285b9289e009830ea
```
Comment (FrankXie05) [inline ports/cubeb/portfile.cmake:4]:
```suggestion
    REF d238150bbacccfb97e62499b0e79a834fcbbd102 # v0.2
```
There is a tag pointing to version 0.2.

File: ports/cubeb/vcpkg.json [hunk 5]
```diff
@@ -0,0 +1,18 @@
+{
+  "name": "cubeb",
+  "version-date": "2022-07-22",
```
Comment (FrankXie05) [inline ports/cubeb/vcpkg.json:3]:
```suggestion
  "version": "0.2",
```

**Review by FrankXie05 — CHANGES_REQUESTED** [review]

File: ports/cubeb/portfile.cmake (hunk 4 above)
Comment (Zopolis4) [reply ports/cubeb/portfile.cmake:4]: Yes, but this tag is from 2012.

Comment (FrankXie05) [reply ports/cubeb/portfile.cmake:4]: We should annotate the specific commit number date for easy updating and maintenance. 
I think this is better. :)

**Review by FrankXie05 — APPROVED** [review]

**Comment by FrankXie05** [comment]
The new usage has been successfully tested locally.
```
cubeb provides CMake targets:

   # this is heuristically generated, and may not be correct 
   find_package(cubeb CONFIG REQUIRED) 
   target_link_libraries(main PRIVATE cubeb::cubeb)
 
````

**Pull request merged**